### PR TITLE
Transaction pool price validation improvements and fixes

### PR DESCRIPTION
**Pull request by fab-10**
## PR description

Review and move all price validation rules in a single method called during the validation of the transaction, instead of having them scattered in different places.

Fix the way the TX_FEECAP_EXCEEDED validation is done, since before it was based on network conditions at the time the transaction was submitted, with the possibility that when the transaction was confirmed the actual fee was different and potentially about the cap. Now the check is done taking in consideration the `maxFeePerGas` that is invariant.

Rename `satisfiesFloorTxCost` to `satisfiesFloorTxFee` since it is about tx fee and not cost.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPoolLondonTest.java [hunk 1]
```diff
@@ -193,17 +194,92 @@ public void shouldAcceptBaseFeeFloorGasPriceFrontierTransactionsWhenMining() {
...
+    final Wei txMaxFeePerGas = minGasPrice.subtract(1L);
+
+    assertThat(
+            add1559TxAndGetPendingTxsCount(
+                genesisBaseFee, minGasPrice, lastBlockBaseFee, txMaxFeePerGas, false))
+        .isEqualTo(0);
+  }
+
+  @Test
+  public void shouldAcceptRemote1559TxsWhenMaxFeePerGasAboveMinGasPrice() {
```
Comment (macfarla) [inline ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPoolLondonTest.java:211]: txMaxFeePerGas = minGasPrice
it's not above but equal - should rename test?

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java [hunk 2]
```diff
@@ -133,45 +132,39 @@ && minTransactionGasPrice(transaction).compareTo(configuration.getTxFeeCap()) >
...
         traceLambda(LOG, "Discard already present transaction {}", transaction::toTraceLog);
         // We already have this transaction, don't even validate it.
         duplicateTransactionCounter.labels(REMOTE).inc();
         continue;
       }
-      final Wei transactionGasPrice = minTransactionGasPrice(transaction);
-      if (transactionGasPrice.compareTo(miningParameters.getMinTransactionGasPrice()) < 0) {
-        traceLambda(
-            LOG,
-            "Discard transaction {} below min gas price {}",
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:166]: do we still need to log this somewhere?

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java [hunk 3]
```diff
@@ -355,6 +337,38 @@ && strictReplayProtectionShouldBeEnforceLocally(chainHeadBlockHeader)
     }
   }
 
+  private TransactionInvalidReason validatePrice(
+      final Transaction transaction, final boolean isLocal, final FeeMarket feeMarket) {
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:341]: it is nice to have this all in one method

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java [hunk 4]
```diff
@@ -375,17 +389,6 @@ private Optional<BlockHeader> getChainHeadBlockHeader() {
...
   }
 
-  private Wei minTransactionGasPrice(final Transaction transaction) {
-    return getChainHeadBlockHeader()
-        .map(
-            chainHeadBlockHeader ->
-                protocolSchedule
-                    .getByBlockNumber(chainHeadBlockHeader.getNumber())
-                    .getFeeMarket()
-                    .minTransactionPriceInNextBlock(transaction, chainHeadBlockHeader::getBaseFee))
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:385]: this minTransactionPriceInNextBlock method doesn't get called anywhere else - do we not need it?

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java (hunk 2 above)
Comment (fab-10) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:166]: readded into `validatePrice` method

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java (hunk 4 above)
Comment (fab-10) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPool.java:385]: good catch removed

File: ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPoolLondonTest.java (hunk 1 above)
Comment (fab-10) [reply ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPoolLondonTest.java]: renamed

**Review by macfarla — APPROVED** [review]

**Pull request merged**